src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -537,6 +551,11 @@ GetDistributedTableDDLEvents(Oid relationId)
 		List *sequenceDDLCommands = SequenceDDLCommandsForTable(relationId);
 		commandList = list_concat(commandList, sequenceDDLCommands);
 
+		/* commands to create user-defined functions appearing in DEFAULT values of a column */
+		List *functionDDLCommands = FunctionDDLCommandsForColumnDefaultsOfTable(
+			relationId);
+		commandList = list_concat(commandList, functionDDLCommands);
```

Comment:
For sequences, extending the metadata syncing logic makes sense to me given that we only ever need to propagate them to workers when using metadata syncing and we have special logic to handle sequences which is different from how we handle all other objects. 

For functions, things are a little fuzzier since the function can also be used in other ways (e.g. queries). We normally propagate functions via the object propagation logic after calling create_distributed_function. It seems kind of odd to have another way of propagating functions here that behaves differently. For instance, we normally propagate CREATE OR REPLACE FUNCTION to the workers via object propagation, but we wouldn't do that if the function was propagated via metadata syncing.

Maybe we leave functions out-of-scope for this PR? We can say that we currently require users to call create_distributed_function if they call a function on the DEFAULT and use metadata syncing.

Reply:
So, for functions, can we do the following for now, for both `create_distributed_table` and `ALTER TABLE ALTER/ADD COLUMN`:
- for non-MX tables, remove the "DEFAULT function_call()" when creating the table shards
- for MX tables, error out and suggest the user to distribute the function first

Reply:
> for MX tables, error out and suggest the user to distribute the function first

makes sense, we currently use separate connections for metadata syncing and function propagation, so it is hard to reconcile these.

Reply:
> for MX tables, error out and suggest the user to distribute the function first

thinking about this more: does this recreate the situation where things fail when we start syncing metadata?

if so, we should force the user to *always* distribute the function first, even for non-MX.